hawc/apps/assessment/models.py

```diff
@@ -737,10 +737,64 @@ def try_read_df(cls, data, suffix: str, worksheet_name: str = None) -> pd.DataFr
         return df
 
 
+class Job(models.Model):
+    PENDING = 1
+    SUCCESS = 2
+    FAILURE = 3
+    STATUS_CHOICES = (
+        (1, "PENDING"),
+        (2, "SUCCESS"),
+        (3, "FAILURE"),
+    )
+
+    TEST = 1
+    JOB_CHOICES = ((1, "TEST"),)
+
+    def test(self, fail=False):
+        if fail:
+            raise Exception("FAILURE")
+        return "SUCCESS"
+
+    task_id = models.UUIDField(primary_key=True, editable=False)
+    assessment = models.ForeignKey(
+        Assessment, null=True, blank=True, on_delete=models.CASCADE, related_name="jobs"
+    )
+    status = models.PositiveSmallIntegerField(
+        choices=STATUS_CHOICES, default=PENDING, editable=False
+    )
+    job = models.PositiveSmallIntegerField(choices=JOB_CHOICES, default=TEST)
+
+    kwargs = JSONField(default=dict)
+    result = models.TextField(blank=True, editable=False)
```

Comment:
both `result` and `exception` fields are TextFields instead of JSONFields, my thought is that this way the task return doesn't have to be in JSON form and it can be parsed on a per job basis. thoughts?

Reply:
i think we can make them JSON fields, if we need to save binary data we'll probably write files and then we can just link to the file in the JSON spec